app/src/main/java/com/moviefiend/torian/moviefiend/MovieDetailsFragment.java

```diff
@@ -1,61 +1,63 @@
 package com.moviefiend.torian.moviefiend;
 
-import android.app.Fragment;
-import android.graphics.Movie;
 import android.os.Bundle;
 import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.ImageView;
-import android.widget.TextView;
 
-import com.moviefiend.torian.moviefiend.network.NowPlayingResponse;
-import com.squareup.picasso.Picasso;
+import com.moviefiend.torian.moviefiend.network.MoviesResponse;
 
-public class MovieDetailsFragment extends Fragment {
+public class MovieDetailsFragment extends Fragment implements MovieDetailsCustomView.Listener {
 
-    public static final int MOVIE_POSTER_SIZE = 780;
+    public interface Listener {
+        void onSimilarMoviesClicked(MoviesResponse.MovieInfo movieInfo);
+    }
+
+    public static final String MOVIE_INFO_ARG = "movie_info";
 
-    private NowPlayingResponse.MovieInfo mMovieInfo;
-    private ImageView mPosterView;
-    private TextView mRatingView;
-    private TextView mDescriptionView;
+    private MoviesResponse.MovieInfo mMovieInfo;
+    private Listener mListener;
 
     @Nullable
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.movie_details_fragment, container, false);
+        MovieDetailsCustomView customView = (MovieDetailsCustomView) view;
 
-        mMovieInfo = getArguments().getParcelable(MovieDetailsActivity.MOVIE_INFO_EXTRA);
-
-        mPosterView = (ImageView) view.findViewById(R.id.poster);
-        mRatingView = (TextView) view.findViewById(R.id.rating);
-        mDescriptionView = (TextView) view.findViewById(R.id.description);
+        mMovieInfo = getArguments().getParcelable(MOVIE_INFO_ARG);
+        customView.setMovie(mMovieInfo);
+        customView.setListener(this);
 
-        setTitle(mMovieInfo.getTitle());
-        setPosterView(UrlHelper.getPosterUrl(MOVIE_POSTER_SIZE, mMovieInfo.getPosterPath()));
-        setRating(mMovieInfo.getRating());
-        setDescription(mMovieInfo.getDescription());
+        if (getUserVisibleHint()) {
```

Comment:
I'm guessing you're using this to detect when the view pager is showing the fragment so you can update the activity's title.

Not crazy about it, since the action bar (and its title) belongs to the activity and so it's the activity who should update it and not an inner level component (the fragment in this case).
Also, who said that the `MovieDetailsFragment` is shown inside a view pager?
Maybe it will be shown in a different container where updating the title based on the user visible hint is not suitable.

It's true that if you move this logic to the activity then you'll have to duplicate it between the `MovieDetailsActivity` and `SimilarMoviesActivity` but at least then you'll be able to use the `ViewPager.OnPageChangeListener` which I think is more reliable than the user visible hint.